### ISSUE TITLE
Suggestion: Make the README.md a lot more detailed

### ISSUE THREAD

**Issue by adriandelgg** [NomicFoundation/hardhat-vscode#159]
Hi team,

Just want to start by saying thank you for creating Hardhat & this extension. They're both amazing!

My suggestion is that the `README.md` needs to be a lot more detailed and documented. It's currently pretty vague and doesn't tell you many ways that you can tweak the extension or more detail on its features and how they work.

Take [Juan's](https://github.com/juanfranblanco/vscode-solidity) `README.md` for example, his is extremely detailed with a lot of screenshots to help the user be able to use the extension better.
Example: How to change the way the extension formats.

A troubleshooting section would also be useful for common issue like the one I had #102.

That's all! Amazing job guys & thank you so much for the work you do for the community!!

**Comment by kanej** [comment]
@adriandelgg agreed. The readme needs expanded out and we definitely need a section on:

* how to setup the formatter
* how to troubleshoot the formatter
* other general troubleshooting
